pyodata/v2/model.py

```diff
@@ -234,6 +234,22 @@ def register_type(typ):
             # pylint: disable=unsupported-assignment-operation
             Types.Types[collection_name] = collection_typ
 
+    @staticmethod
+    def overwrite_type(typ):
+        """Add new  type to the type repository as well as its collection variant"""
+
+        # build types hierarchy on first use (lazy creation)
+        if Types.Types is None:
+            Types._build_types()
+
+        # pylint: disable=unsupported-assignment-operation
+        Types.Types[typ.name] = typ
+
+        collection_name = f'Collection({typ.name})'
```

Comment:
Keep in mind this commit
https://github.com/SAP/python-pyodata/commit/bfa215f138d7e76737fb6d2e268a513e30a7ba84